File: Systems-for-Design-and-Implementation/Labs/Lab12/ProiectJava/TransportNetworking/src/main/java/transport/network/jsonprotocol/JsonProtocolUtils.java
package transport.network.jsonprotocol;

import transport.model.Reservation;
import transport.model.ReservationManager;
import transport.model.Seat;
import transport.model.Trip;

import java.util.List;

public class JsonProtocolUtils {
    public static Response createNewReservationResponse(Reservation reservation){
        Response resp=new Response();
        resp.setType(ResponseType.NEW_RESERVATION);
        resp.setReservation(reservation);
        return resp;
    }

    public static Response createOkResponse(){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        return resp;
    }

    public static Response createErrorResponse(String errorMessage){
        Response resp=new Response();
        resp.setType(ResponseType.ERROR);
        resp.setErrorMessage(errorMessage);
        return resp;
    }

    public static Response createGetAllTripsResponse(Trip[] trips){
        Response resp=new Response();
        resp.setType(ResponseType.GET_ALL_TRIPS);
        resp.setTrips(trips);
        return resp;
    }

    public static Response createGetSeatsByTripResponse(List<Seat> seats){
        Response resp=new Response();
        resp.setType(ResponseType.GET_SEATS_BY_TRIP);
        resp.setSeats(seats);
        return resp;
    }

    public static Response createSearchTripByIdResponse(Trip trip){
        Response resp=new Response();
        resp.setType(ResponseType.SEARCH_TRIP_BY_ID);
        resp.setTrip(trip);
        return resp;
    }

    public static Response createGetReservationsByTripResponse(List<Reservation> reservations){
        Response resp=new Response();
        resp.setType(ResponseType.GET_RESERVATIONS_BY_TRIP);
        resp.setReservations(reservations);
        return resp;
    }

    public static Response createGetSeatsByReservationResponse(List<Seat> seats){
        Response resp=new Response();
        resp.setType(ResponseType.GET_SEATS_BY_RESERVATION);
        resp.setSeats(seats);
        return resp;
    }

    public static Request createLoginRequest(ReservationManager reservationManager){
        Request req=new Request();
        req.setType(RequestType.LOGIN);
        req.setReservationManager(reservationManager);
        return req;
    }

    public static Request createSearchReservationManagerByNameRequest(String name){
        Request req=new Request();
        req.setType(RequestType.SEARCH_RESERVATION_MANAGER_BY_NAME);
        req.setStringR(name);
        return req;
    }

    public static Request createGetAllTripsRequest(){
        Request req=new Request();
        req.setType(RequestType.GET_ALL_TRIPS);
        return req;
    }

    public static Request createSearchTripByIdRequest(Long id){
        Request req=new Request();
        req.setType(RequestType.SEARCH_TRIP_BY_ID);
        req.setLong(id);
        return req;
    }

    public static Request createAddReservationRequest(Reservation reservation){
        Request req=new Request();
        req.setType(RequestType.NEW_RESERVATION);
        req.setReservation(reservation);
        return req;
    }

    public static Request createGetReservationsByTripRequest(Trip trip){
        Request req=new Request();
        req.setType(RequestType.GET_RESERVATIONS_BY_TRIP);
        req.setTrip(trip);
        return req;
    }

    public static Request createGetSeatsByReservationRequest(Long reservationId){
        Request req=new Request();
        req.setType(RequestType.GET_SEATS_BY_RESERVATION);
        req.setLong(reservationId);
        return req;
    }

    public static Request createGetSeatsByTripRequest(Trip trip){
        Request req=new Request();
        req.setType(RequestType.GET_SEATS_BY_TRIP);
        req.setTrip(trip);
        return req;
    }

    public static Request createLogoutRequest(ReservationManager reservationManager){
        Request req=new Request();
        req.setType(RequestType.LOGOUT);
        req.setReservationManager(reservationManager);
        return req;
    }
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab8/ProiectCSharp/TransportNetworking/transport.network/jsonprotocol/JsonProtocolUtils.cs
using System.Collections.Generic;
using Transport.Model.transport.model;
using TransportModel.transport.model;

namespace TransportNetworking.transport.network.jsonprotocol
{
    public static class JsonProtocolUtils
    {
        public static Response CreateNewReservationResponse(Reservation reservation)
        {
            return new Response
            {
                Type = ResponseType.NEW_RESERVATION,
                Reservation = reservation
            };
        }

        public static Response CreateOkResponse()
        {
            return new Response
            {
                Type = ResponseType.OK
            };
        }

        public static Response CreateErrorResponse(string errorMessage)
        {
            return new Response
            {
                Type = ResponseType.ERROR,
                ErrorMessage = errorMessage
            };
        }

        public static Response CreateGetAllTripsResponse(Trip[] trips)
        {
            return new Response
            {
                Type = ResponseType.GET_ALL_TRIPS,
                Trips = trips
            };
        }
        
        public static Response CreateGetSeatsByTripResponse(List<Seat> seats)
        {
            return new Response
            {
                Type = ResponseType.GET_SEATS_BY_TRIP,
                Seats = seats
            };
        }

        public static Response CreateSearchTripByIdResponse(Trip trip)
        {
            return new Response
            {
                Type = ResponseType.SEARCH_TRIP_BY_ID,
                Trip = trip
            };
        }
        
        public static Response CreateGetReservationsByTripResponse(List<Reservation> reservations)
        {
            return new Response
            {
                Type = ResponseType.GET_RESERVATIONS_BY_TRIP,
                Reservations = reservations
            };
        }
        
        public static Response CreateGetSeatsByReservationResponse(List<Seat> seats)
        {
            return new Response
            {
                Type = ResponseType.GET_SEATS_BY_RESERVATION,
                Seats = seats
            };
        }

        public static Request CreateLoginRequest(ReservationManager reservationManager)
        {
            return new Request
            {
                Type = RequestType.LOGIN,
                ReservationManager = reservationManager
            };
        }

        public static Request CreateSearchReservationManagerByNameRequest(string name)
        {
            return new Request
            {
                Type = RequestType.SEARCH_RESERVATION_MANAGER_BY_NAME,
                String = name
            };
        }

        public static Request CreateGetAllTripsRequest()
        {
            return new Request
            {
                Type = RequestType.GET_ALL_TRIPS
            };
        }

        public static Request CreateSearchTripByIdRequest(long id)
        {
            return new Request
            {
                Type = RequestType.SEARCH_TRIP_BY_ID,
                Long = id
            };
        }

        public static Request CreateAddReservationRequest(Reservation reservation)
        {
            return new Request
            {
                Type = RequestType.NEW_RESERVATION,
                Reservation = reservation
            };
        }

        public static Request CreateGetReservationsByTripRequest(Trip trip)
        {
            return new Request
            {
                Type = RequestType.GET_RESERVATIONS_BY_TRIP,
                Trip = trip
            };
        }

        public static Request CreateGetSeatsByReservationRequest(long reservationId)
        {
            return new Request
            {
                Type = RequestType.GET_SEATS_BY_RESERVATION,
                Long = reservationId
            };
        }

        public static Request CreateGetSeatsByTripRequest(Trip trip)
        {
            return new Request
            {
                Type = RequestType.GET_SEATS_BY_TRIP,
                Trip = trip
            };
        }

        public static Request CreateLogoutRequest(ReservationManager reservationManager)
        {
            return new Request
            {
                Type = RequestType.LOGOUT,
                ReservationManager = reservationManager
            };
        }
    }
}